curriculum/challenges/english/02-javascript-algorithms-and-data-structures/debugging/catch-off-by-one-errors-when-using-indexing.md

```diff
@@ -49,7 +49,7 @@ assert(!code.match(/i\s?=\s*?1\s*?;/g));
 Your code should set the terminal condition of the loop so it stops at the last index.
 
 ```js
-assert(code.match(/i\s*?<\s*?len\s*?;/g).length == 1);
+assert(code.match(/i\s*<\s*len\s*;|i\s*<=\s*len\s*-\s*1\s*;/g).length == 1);
```

Comment:
```suggestion
assert(code.match(/i\s*<=?\s*len\s*-?\s*1?;/g).length == 1);
```

Reply:
@Sboonny I was also thinking to use the optional matching but it may allow some invalid tests too.

Valid tests-
`i < len;`
`i <= len-1;`

Invalid tests-
`i <= len;`
`i <= len-;`
`i <= len1;`
...etc

Reply:
Good catch 👍